homeassistant/components/husqvarna_automower/sensor.py

```diff
@@ -0,0 +1,173 @@
+    AutomowerSensorEntityDescription(
+        key="total_searching_time",
+        translation_key="total_searching_time",
+        icon="mdi:clock-outline",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        state_class=SensorStateClass.TOTAL,
+        device_class=SensorDeviceClass.DURATION,
+        native_unit_of_measurement=UnitOfTime.SECONDS,
+        suggested_unit_of_measurement=UnitOfTime.HOURS,
+        value_fn=lambda data: data.statistics.total_searching_time,
+    ),
+    AutomowerSensorEntityDescription(
+        key="number_of_charging_cycles",
+        translation_key="number_of_charging_cycles",
+        icon="mdi:battery-sync-outline",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        state_class=SensorStateClass.TOTAL,
+        value_fn=lambda data: data.statistics.number_of_charging_cycles,
+    ),
+    AutomowerSensorEntityDescription(
+        key="number_of_collisions",
+        translation_key="number_of_collisions",
+        icon="mdi:counter",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        state_class=SensorStateClass.TOTAL,
+        value_fn=lambda data: data.statistics.number_of_collisions,
+    ),
+    AutomowerSensorEntityDescription(
+        key="total_drive_distance",
+        translation_key="total_drive_distance",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        state_class=SensorStateClass.TOTAL,
+        device_class=SensorDeviceClass.DISTANCE,
+        native_unit_of_measurement=UnitOfLength.METERS,
+        suggested_unit_of_measurement=UnitOfLength.KILOMETERS,
+        value_fn=lambda data: data.statistics.total_drive_distance,
+    ),
+    AutomowerSensorEntityDescription(
+        key="next_start_timestamp",
+        translation_key="next_start_timestamp",
+        device_class=SensorDeviceClass.TIMESTAMP,
+        value_fn=lambda data: data.planner.next_start_dateteime,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up sensor platform."""
+    coordinator: AutomowerDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
+    async_add_entities(
+        AutomowerSensorEntity(mower_id, coordinator, description)
+        for mower_id in coordinator.data
+        for description in SENSOR_TYPES
+        if description.exists_fn(coordinator.data[mower_id])
+    )
+
+
+class AutomowerSensorEntity(AutomowerBaseEntity, SensorEntity):
+    """Defining the Automower Sensors with AutomowerSensorEntityDescription."""
+
+    entity_description: AutomowerSensorEntityDescription
+
+    def __init__(
+        self,
+        mower_id: str,
+        coordinator: AutomowerDataUpdateCoordinator,
+        description: AutomowerSensorEntityDescription,
+    ) -> None:
+        """Set up AutomowerSensors."""
+        super().__init__(mower_id, coordinator)
+        self.entity_description = description
+        self._attr_unique_id = f"{self.mower_id}_{description.key}"
```

Comment:
```suggestion
        self._attr_unique_id = f"{mower_id}_{description.key}"
```

Reply:
done in https://github.com/home-assistant/core/pull/110410/commits/63e53a206dc05a5fa27423df05ac7dedaf4b4ab2